pkg/repositories/repositories.go

```diff
@@ -57,6 +57,44 @@ func NewConfig(filename string) (Config, error) {
 	return repoConfig, nil
 }
 
+// If our first config and our second config share a repository,
+// we set the afterVersion of the first to its original version, and the
+// version of the first to the version of the second.
+func CombineConfigs(currentConfig *Config, newConfig *Config) {
+
+	// Extract new repos regardless of category
+	var newRepos []Repository
+	for _, category := range newConfig.Section.Categories {
+		newRepos = append(newRepos, category.Repos...)
+	}
+
+	// Iterate through current categories and repositories, maintaining
+	// the structure.
+	for categoryIdx := range currentConfig.Section.Categories {
+		for repoIdx, currentRepo := range currentConfig.Section.Categories[categoryIdx].
+			Repos {
```

Comment:
When reading through this code, this line looks like you're creating a `Repo{}` object and not that it's part of the `for` loop. Some clarification might be needed here to make it apparent that it's the latter.

Reply:
Yeah my hard line wraps messed this up - fixed